contracts/src/ethereum/TokenBridge.sol

```diff
@@ -79,29 +82,32 @@ contract TokenBridge is
         l2TokenBridge = l2TokenBridge_;
     }
 
-    modifier isValidL2Address(uint256 l2Address) {
+    modifier isValiduint256(uint256 l2Address) {
         require((l2Address != 0) && (l2Address < CairoConstants.FIELD_PRIME), "L2_ADDRESS_OUT_OF_RANGE");
         _;
     }
 
-    function approveBridge(address l1Token, uint256 l2Token)
+    function approveBridge(IStaticATokenLM l1Token, uint256 l2Token)
         external
         onlyGovernance
-        isValidL2Address(l2Token)
+        isValiduint256(l2Token)
     {
-        require(l1Token != address(0x0), "l1Token address cannot be 0x0");
+        require(l1Token != IStaticATokenLM(0x0), "l1Token address cannot be 0x0");
 
         uint256 l2Token_ = l1TokentoL2Token[l1Token];
         require(l2Token_ == 0, "l2Token already set");
 
+        l1Token.ASSET().safeApprove(address(l1Token), type(uint256).max);
```

Comment:
why adding these approvals? do they simplify the flow on the e2e test?

Reply:
We do this because we call deposit on the l1Token in  the `depositUnderlying` function. Without approving the staticAToken as a spender of the assets owned by the bridge we would not be able to deposit. It's not relevant anymore for the changes that are coming.